thirdeye/thirdeye-frontend/app/pods/rootcause-aggregates-cache/service.js

```diff
@@ -56,12 +56,43 @@ export default Service.extend({
       return;
     }
 
-    // metrics
+    let metricUrnToOffestAndUrn = {};
```

Comment:
also we can use a `const` here. const objects can have their property change/updated/deleted but the reference to the original object must remain intact. 
`metricUrnToOffestAndUrn[metricUrn] = [];` will not be allowed, but if you apply my next comment, this should be a non issue